presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -92,7 +92,7 @@ public void run()
         boolean hasQuery = !Strings.isNullOrEmpty(clientOptions.execute);
         boolean isFromFile = !Strings.isNullOrEmpty(clientOptions.file);
 
-        if (!hasQuery || !isFromFile) {
+        if (!hasQuery && !isFromFile) {
```

Comment:
The relevant part of the code in `AnsiConsole` is below and it calls `isatty`, which is a native call to the `isatty` syscall. And if the stdout is redirected to some file then `isatty` returns 0, causing `AnsiOutputStream` to be installed.

```
int rc = isatty(STDOUT_FILENO);
if( rc==0 ) {
    return new AnsiOutputStream(stream);
}
```